SYCL/ESIMD/api/functional/functions/functions_1d_select.hpp

```diff
@@ -0,0 +1,281 @@
+
+    for (size_t i = 0; i < NumSelectedElems; ++i) {
+      select_result[i] = ref_2[i];
+    }
+    simd_obj.copy_to(out);
+
+    return std::is_same_v<
+        decltype(select_result),
+        simd_view<esimd::simd<DataT, NumElems>,
+                  region1d_t<DataT, NumSelectedElems, Stride>>>;
+  }
+};
+
+// Descriptor class for the case of calling simd<T,N>::select function.
+struct select_lval {
+  static std::string get_description() { return "select lvalue"; }
+
+  template <typename DataT, int NumElems, int NumSelectedElems, int Stride>
+  static bool call_operator(const DataT *const ref_1, const DataT *const ref_2,
+                            DataT *const out, size_t offset) {
+    simd<DataT, NumElems> src_simd_obj;
+    src_simd_obj.copy_from(ref_1);
+
+    simd<DataT, NumSelectedElems> dst_simd_obj;
+    dst_simd_obj.copy_from(ref_2);
+
+    src_simd_obj.template select<NumSelectedElems, Stride>(offset) =
+        dst_simd_obj;
+    src_simd_obj.copy_to(out);
+
+    return true;
+  }
+};
+
+// Descriptor class for the case of calling simd<T,N>::select function.
+struct select_simd_view {
+  static std::string get_description() { return "select simd view"; }
+
+  template <typename DataT, int NumElems, int NumSelectedElems, int Stride>
+  static bool call_operator(const DataT *const ref_1, const DataT *const ref_2,
+                            DataT *const out, size_t offset) {
+    simd<DataT, NumElems> src_simd_obj;
+    src_simd_obj.copy_from(ref_1);
+
+    auto simd_view_instance = src_simd_obj.template bit_cast_view<DataT>();
+
+    auto selected_elems =
+        simd_view_instance.template select<NumSelectedElems, Stride>(offset);
+
+    for (size_t i = 0; i < NumSelectedElems; ++i) {
+      selected_elems[i] = ref_2[i];
+    }
+
+    src_simd_obj.copy_to(out);
+
+    return true;
+  }
+};
+
+// The main test routine.
+// Using functor class to be able to iterate over the pre-defined data types.
+template <typename TestCaseT, typename NumSelectedElemsT, typename StrideT,
+          typename OffsetT, typename DataT, typename DimT>
+class run_test {
+  static constexpr int NumElems = DimT::value;
+  static constexpr int NumSelectedElems = NumSelectedElemsT::value;
+  static constexpr int Stride = StrideT::value;
+  static constexpr int Offset = OffsetT::value;
+
+public:
+  bool operator()(sycl::queue &queue, const std::string &data_type) {
+    static_assert(NumElems >= NumSelectedElems * Stride + Offset &&
+                  "Offset should be less than number selected elements.");
+    bool passed = true;
+    size_t alignment_value = alignof(DataT);
+
+    constexpr size_t value_for_increase_ref_data_for_fill = 50;
+    static_assert(std::numeric_limits<signed char>::max() >
+                      value_for_increase_ref_data_for_fill + NumElems,
+                  "Value that used for increase ref data for fill plus simd "
+                  "size should be less than char max value.");
```

Comment:
```suggestion
    constexpr size_t value_for_increase_ref_data_for_fill = 50;
    static_assert(std::numeric_limits<signed char>::max() >
                      value_for_increase_ref_data_for_fill + NumElems);
```
No need to duplicate information. Looks like it's easy to forget update it in two places.

Reply:
this comment has been applied [here](https://github.com/intel/llvm-test-suite/pull/875/commits/03d94c35b4a95e3ce26c99e1197bc50ccdc652aa#diff-295141c99d991cbb4971abc3bd754b2f17400cade3b579ac7bc36b28a1450dc3L130-R136)